Assets/Scripts/Gameplay/TurnController.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Gameplay
+{
+    public class TurnController : MonoBehaviour
+    {
+        [SerializeField]
+        public int currentTurn = 0;
+
+        //private Dictionary<int, Player> playerList;
+        private List<Player> turnOrder = new();
+
+        public void AddPlayer(Player player)
+        {
+            if (turnOrder.Contains(player))
+                return;
+
+            turnOrder.Add(player);
+        }
+
+        public void ResetTurnOrder()
+        {
+            turnOrder = (List<Player>)turnOrder.Shuffle();
+            currentTurn = 0;
+        }
+
+        public Player CurrentPlayer()
+        {
+            if (turnOrder.Count < 1)
+            {
+                return null;
```

Comment:
I'm not really sure about this but in java it'd be idiomatic to use an optional here. It looks like C# has them too https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.optional-1?view=roslyn-dotnet-4.2.0